File: app/src/main/java/com/arbaelbarca/listmovieskatalog/presenter/detailmoviespresenter/DetailMoviesPresenterImpl.java
package com.arbaelbarca.listmovieskatalog.presenter.detailmoviespresenter;

import android.content.Context;

import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.ResponseDetailMovies;
import com.arbaelbarca.listmovieskatalog.model.modelreviewmovies.ResponseReviewMovies;
import com.arbaelbarca.listmovieskatalog.network.NetworkApi;
import com.arbaelbarca.listmovieskatalog.view.DetailMoviesView;

import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class DetailMoviesPresenterImpl implements DetailMoviesView.GetNoticeIntractorDetailMovies, DetailMoviesView.presenter {

    private Context context;

    public DetailMoviesPresenterImpl(Context context) {
        this.context = context;
    }

    @Override
    public void getDetailMovies(OnFinishedListener onFinishedListener, String movie_id) {
        NetworkApi
                .getInstance()
                .getAPI()
                .getDetailMovies(movie_id, "en-US")
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Subscriber<ResponseDetailMovies>() {
                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {

                    }

                    @Override
                    public void onNext(ResponseDetailMovies responseDetailMovies) {
                        if (responseDetailMovies != null) {
                            onFinishedListener.onFinished(responseDetailMovies);
                        }
                    }
                });
    }

    @Override
    public void getDetailReviews(OnFinishedListener onFinishedListener, String movie_id) {
        NetworkApi
                .getInstance()
                .getAPI()
                .getDetailReviews(movie_id, "en-US")
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Subscriber<ResponseReviewMovies>() {
                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {

                    }

                    @Override
                    public void onNext(ResponseReviewMovies responseReviewMovies) {
                        if (responseReviewMovies.getResults() != null) {
                            onFinishedListener.setDataReview(responseReviewMovies.getResults());
                        }
                    }
                });
    }


    @Override
    public void onDestroy() {

    }

    @Override
    public void requestFromDataServer(String movieId) {

    }


    @Override
    public void refreshData(String type) {

    }
}

File: app/src/main/java/com/arbaelbarca/listmovieskatalog/presenter/detailmoviespresenter/DetailMoviesPresenterImpl.kt
package com.arbaelbarca.listmovieskatalog.presenter.detailmoviespresenter

import android.content.Context

import com.arbaelbarca.listmovieskatalog.model.modeldetailmovies.ResponseDetailMovies
import com.arbaelbarca.listmovieskatalog.model.modelreviewmovies.ResponseReviewMovies
import com.arbaelbarca.listmovieskatalog.network.NetworkApi
import com.arbaelbarca.listmovieskatalog.utils.Constants
import com.arbaelbarca.listmovieskatalog.view.DetailMoviesView

import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class DetailMoviesPresenterImpl(private val context: Context) : DetailMoviesView.GetNoticeIntractorDetailMovies, DetailMoviesView.presenter {

    override fun getDetailMovies(onFinishedListener: DetailMoviesView.GetNoticeIntractorDetailMovies.OnFinishedListener, movie_id: String) {
        NetworkApi
                .instance
                .api
                .getDetailMovies(movie_id, Constants.LANGUAGE)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : Subscriber<ResponseDetailMovies>() {
                    override fun onCompleted() {

                    }

                    override fun onError(e: Throwable) {

                    }

                    override fun onNext(responseDetailMovies: ResponseDetailMovies?) {
                        if (responseDetailMovies != null) {
                            onFinishedListener.onFinished(responseDetailMovies)
                        }
                    }
                })
    }

    override fun getDetailReviews(onFinishedListener: DetailMoviesView.GetNoticeIntractorDetailMovies.OnFinishedListener, movie_id: String) {
        NetworkApi
                .instance
                .api
                .getDetailReviews(movie_id, Constants.LANGUAGE)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : Subscriber<ResponseReviewMovies>() {
                    override fun onCompleted() {

                    }

                    override fun onError(e: Throwable) {

                    }

                    override fun onNext(responseReviewMovies: ResponseReviewMovies) {
                        if (responseReviewMovies.results != null) {
                            onFinishedListener.setDataReview(responseReviewMovies.results)
                        }
                    }
                })
    }


    override fun onDestroy() {

    }

    override fun requestFromDataServer(movieId: String) {

    }


    override fun refreshData(type: String) {

    }
}
